### ISSUE TITLE
Captcha issue when applying as a reviewer

### ISSUE THREAD

**Issue by MonicaMuranyi** [appirio-tech/topcoder-app#917]
Every time I want to apply as a reviewer in a challenge I have to add the captcha code multiple times although I enter the correct word. I often get an error page and then I have to click the browser back button and start again entering the captchas until it works. This doesn't happen consistently but it happens in 70% of cases.

**Comment by birdofpreyru** [comment]
@ajefts Is there any way to turn a test user into a reviewer, so it can be tested and fixed?

**Comment by ajefts** [comment]
yes.  I just made reviewer1/topcoder#1 as a code, f2f, and ui proto reviewer.

**Comment by birdofpreyru** [comment]
@ajefts @MonicaMuranyi Should I see the captcha at the same page, where I see the open review positions, and have the `Apply Now` button? As when I try test reviewer on www.topcoder-dev.com, I don't see it there, but also it does not allow me to press `Apply Now` - nothing happens.

@ajefts Also, which repo hosts the reviewer auction code?

**Comment by birdofpreyru** [comment]
### Expected Fix
1. Find out, how to test for this issue with the locally deployed App (keep in mind, that the related code, might be in another repo).
2. Fix the issue, and provide brief comments, how to test.

**Comment by Colinh84** [comment]
@birdofpreyru Can I take this issue please? For some reason, I am unable to assign myself..

**Comment by birdofpreyru** [comment]
@Colinh84 Sure, it is yours. I'll also update your rights in this repo in a moment

**Comment by Colinh84** [comment]
@birdofpreyru Thanks!

**Comment by Colinh84** [comment]
@birdofpreyru I'm not sure this one can be solved - The review system is not part of this repo, and there is no captcha related code either...

**Comment by birdofpreyru** [comment]
@Colinh84 Well, you are not restricted to this repo only. The code should be in some of the repos belonging to `appirio-tech`, I believe all they are public, so you can look around, to see if you can find it.

**Comment by Colinh84** [comment]
@birdofpreyru I've found the repo responisble which is `tc1-tcnode`. However, I'm having trouble getting it deployed properly on my local linux environment, so I will let someone try this.

**Comment by MonicaMuranyi** [comment]
The captcha only appears after clicking the "Apply Now" button. Sorry for the late reply on this.

**Comment by arifnyet** [comment]
@birdofpreyru can i pick up this ticket ?

**Comment by phinguyen712** [comment]
@birdofpreyru may I have permission to pick up this ticket also?

**Comment by birdofpreyru** [comment]
@arifnyet @phinguyen712 
Well, @arifnyet was first, so the ticket goes to him for now.

**Comment by arifnyet** [comment]
i dont think i can solve this in 24 hrs... i need to solve the js file error first before i can test the captcha.... feel free to remove me if somebody else can solve this faster...

**Comment by birdofpreyru** [comment]
@arifnyet Well, if you want to, feel free to play with this one a bit more, if @phinguyen712 does not mind.
Actually, this one is a bit tricky, as probably demands to deploy the backend code, which is java based... I have not managed to do it myself yet :/

**Comment by arifnyet** [comment]
@birdofpreyru perhaps if someone can upload the latest newTCScript.js located in ./tc1-tcnode/tc-wars/static/src/main/webapp/js/reskin/newTCScript.js it could save me a lot of time of setting up multiple backends.. as you can see the error on the console said at line 823 the value is null which in the old script
`searchNeedle = searchBox.value,`
new script
`searchNeedle = searchBox ? searchBox.value : null,`
thus i believe would solve the js error so i can test the captcha

**Comment by birdofpreyru** [comment]
@arifnyet Nope, at the moment I am really not sure what are you talking about. I have not managed to run that `tc1-tcnode` code myself yet :( @ajefts should be able to help, if he has time.

**Comment by arifnyet** [comment]
@birdofpreyru i dont think i manage to do this.. even with the backend deployed it still give me TCWebException

**Comment by birdofpreyru** [comment]
@arifnyet Ok, no problem. How did you deploy it actually? Just following the README.md, or are there any know-how not mentioned in there?

**Comment by arifnyet** [comment]
@birdofpreyru i use ec2 instance... run a clean ubuntu install and apt-get the needed tech from there... for the informix i have to upload the image manually.... but dont take my word for it as even tho it is deployed it still gives me TCWebException whenever i enter the ec2ip/tc?module=ReviewAuctionApply&aid=12316&arid=9&b=Apply+Now

**Comment by birdofpreyru** [comment]
@arifnyet And where have you got the Informix image from?

**Comment by arifnyet** [comment]
@birdofpreyru https://www-01.ibm.com/marketing/iwm/tnd/preconfig.jsp?id=2013-10-02+10%3A49%3A20.723369R&S_TACT=&S_CMP=

**Comment by birdofpreyru** [comment]
@arifnyet Ok, I've also got Informix from that site.

Currently, when I build everything following instructions from the README, I get `HTTP Status 404` when I try to access `/tc` route :( Still trying to figure out, what am I missing.

**Comment by arifnyet** [comment]
@birdofpreyru ok i check back my setup... i changed a few apache2 setup while doing this

**Comment by birdofpreyru** [comment]
@arifnyet Yeap, I'll appreciate if you help me with it.
So far what I can do:
1. Access `localhost` which shows me JBoss welcome screen.
2. From there I can access `localhost/status`, which says JBoss is fine.
Everything else responds to me with 404 error.

**Comment by arifnyet** [comment]
@birdofpreyru sry for the wait... here is what i do to deploy some of the ear files eventho it is still tcwebexception
[httpd.txt](https://github.com/appirio-tech/topcoder-app/files/743600/httpd.txt)
[steps.txt](https://github.com/appirio-tech/topcoder-app/files/743604/steps.txt)

i suspect the database connection is at fault...

**Comment by birdofpreyru** [comment]
@arifnyet Many thanks! I didn't know that JBoss server can be run with different configs, thus was running the wrong one, calling `run.sh` without `-c all` parameter (well, I never heard about JBoss before at all :)

Now I can see TC error message when I access to `/tc` and a bunch of error messages in JBoss log. Will try to resolve them :)

**Comment by arifnyet** [comment]
@birdofpreyru i leave it to you then... custom made java is hard for me to deal with without racking my brains out...
